alg/CATENets/catenets/datasets/dataset_twins.py

```diff
@@ -0,0 +1,161 @@
+# stdlib
+import random
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import catenets.logger as log
+
+# third party
+import numpy as np
+from scipy.special import expit
+
+from .network import download_if_needed
+
+np.random.seed(0)
+random.seed(0)
+
+DATASET = "Twin_Data.csv.gz"
+URL = "https://bitbucket.org/mvdschaar/mlforhealthlabpub/raw/0b0190bcd38a76c405c805f1ca774971fcd85233/data/twins/Twin_Data.csv.gz"  # noqa: E501
+
+
+def preprocess(fn_csv: Path) -> List[np.ndarray]:
+    """
+    Load the dataset and preprocess it.
+    Input:
+        fn_csv: Path to the input CSV to load
+    Outputs:
+        X: Feature Vector
+        T: Treatment Vector
+        Y: Observable Outcomes
+        Opt_Y: Potential Outcomes
+    """
+    # Data Input (11400 patients, 30 features, 2 potential outcomes)
+    Data = np.loadtxt(fn_csv, delimiter=",", skiprows=1)
+
+    # Features
+    X = Data[:, :30]
+
+    # Feature dimensions and patient numbers
+    Dim = len(X[0])
+    No = len(X)
+
+    # Labels
+    Opt_Y = Data[:, 30:]
+
+    for i in range(2):
+        idx = np.where(Opt_Y[:, i] > 365)
+        Opt_Y[idx, i] = 365
+
+    Opt_Y = 1 - (Opt_Y / 365.0)
```

Comment:
Don't we usually use the Twins dataset with binary outcomes? (i.e. outcome =  (death prior to 365 days == True) ?)

Reply:
The code is copy-paste from Jinsung's preprocessing for GANITE. (I added the credits now).

This version of the dataset works fine with CATENets. This version does not https://github.com/vanderschaarlab/medicaldata/blob/main/medicaldata/Twins/dataloader.py